app/models/video.rb

```diff
@@ -8,4 +9,13 @@ def self.search_by_title(search_term)
     return [] if search_term.blank?
     where("title ILIKE ?", "%#{search_term}%").order("title ASC")
   end
+  
+  def average_rating
+    return 0 if reviews.empty?
+    ratings = []
+    reviews.each do |review|
+      ratings << review.rating
+    end
```

Comment:
Lines 15 to 18 could be refactored to `ratings = reviews.map { |review| review.rating }`

and even further to:

`ratings = reviews.map(&:rating)`